### DISCUSSION TITLE
DataGrid selection mode

### DISCUSSION THREAD

**Discussion by AlexaBuff** [Megabit/Blazorise#2667] [General]
After update 0.9.4 dataGrid's selection mode became multiple automatically. I can not turn on single selection mode. I noticed that multiple selection throw off after double click. My code:

```cshtml
 <DataGrid TItem="Employee" @ref="dataGrid"
              Data="@employeeList"
              Editable="true"
              CommandMode="DataGridCommandMode.ButtonRow"
              ShowPageSizes="true"
              EditMode="DataGridEditMode.Popup"
              Bordered="true"
              Hoverable="true"
              Responsive="true"
              ShowPager="true"
              RowInserted="@CreateEmployee"
              PopupSize="ModalSize.Large"
              RowDoubleClicked="@ShowModal"
              UseInternalEditing="true"
              PagerPosition="DataGridPagerPosition.Top"
              Resizable="true"           
              FixedHeader="true"
              FixedHeaderDataGridHeight="80%"
              ResizeMode="TableResizeMode.Columns"
              SelectionMode="DataGridSelectionMode.Single" >
```

**Comment by stsrki** [comment]
@David-Moreira can you look at this?

**Comment by stsrki** [comment]
I figured what is wrong. You're missing `@bind-SelectedRow="@selectedEmployee"`. If you add it, it will work.

@David-Moreira I think it should also work without the need for `@bind-`. We might look into it.

**Reply by AlexaBuff** [reply]
Ooo, Thank you very much. It worked!

**Reply by David-Moreira** [reply]
> I figured what is wrong. You're missing `@bind-SelectedRow="@selectedEmployee"`. If you add it, it will work.
> 
> @David-Moreira I think it should also work without the need for `@bind-`. We might look into it.

Not quite being in multi select. Just that the rows aren't re-evaluating the stylying, as they think nothing has changed. The @bind-SelectedRow solves this, as it binds an EventCallback behind the scenes, and that takes care of calling StateHasChanged.
I'll confirm this later today, and fix. 
We've made quite some performance improvements on datagrid and it was quite the change, so bear with us if we didn't get right some things. We'll fix it asap.
